front/src/statistics/Statistics.js

```diff
@@ -0,0 +1,385 @@
+    if (event?.event.type?.event.type === 'submit') {
+      event.preventDefault();
+    }
+  };
+
+  const countMonthlyVisitors = () => monthlyUsers.reduce((a, b) => a + b, 0);
+
+  const handleDatePickChange = (newDate) => {
+    const newDateObject = new Date(newDate);
+    setDate(newDateObject);
+    continueWithDate();
+  };
+
+  if (monthlyUsers?.length > 0) {
+    const total = countMonthlyVisitors();
+
+    return (
+      <div className="container">
+        {/* Section for selecting date */}
+        <div className="firstSection">
+          <form onSubmit={continueWithDate}>
+            { /* Datepicker */}
+            <MuiPickersUtilsProvider
+              utils={MomentUtils}
+              locale={lang}
+            >
+              <KeyboardDatePicker
+                autoOk
+                margin="normal"
+                name="date"
+                label={statistics.DayChoose[fin]}
+                value={date}
+                onChange={(newDate) => handleDateChange(newDate)}
+                onAccept={handleDatePickChange}
+                format="DD.MM.YYYY"
+                showTodayButton
+              />
+            </MuiPickersUtilsProvider>
+          </form>
+        </div>
+        <hr />
+
+        {/* Header with arrows */}
+        <Grid class="date-header">
+          <div
+            className="hoverHand arrow-left"
+            onClick={previousDayClick}
+          />
+          <h1 className="headerText">
+            {date.toLocaleDateString('fi-FI')}
+          </h1>
+          <div
+            className="hoverHand arrow-right"
+            onClick={nextDayClick}
+          />
+        </Grid>
+        {/* Charts */}
+        <div className="row">
+          <div className="mixed-chart">
+            {/* Labels */}
+            <h2>{statistics.Day[fin]}</h2>
+            <h3>
+              {`${statistics.Total[fin]} ${date.toLocaleDateString('fi-FI')}: ${monthlyUsers[dayNumber - 1]}`}
+            </h3>
+            <div className="bar">
+              <Chart
+                options={dayOptions}
+                series={daySeries}
+                type="bar"
+                width="700"
+                height="400"
+              />
+            </div>
+            {/* Labels */}
+            <h2>{statistics.Month[fin]}</h2>
+            <h3>
+              {`${statistics.Total[fin]} ${(date.getMonth() + 1)}/${date.getFullYear()}: ${total}`}
+            </h3>
+            <div className="line">
+              <Chart
+                options={monthOptions}
+                series={monthSeries}
+                type="line"
+                width="700"
+                height="400"
+              />
+            </div>
+            <div className="bar">
+              <Chart
+                options={monthlyTrackOptions}
+                series={monthlyTrackSeries}
+                type="bar"
+                width="700"
+                height="400"
+              />
+            </div>
```

Comment:
Track labels should be by short_description, not id.

Reply:
No idea how am I going to do that properly. I made another API call but I think it's quite clumsy solution to get the short descriptions of each track.